rpcs3/Emu/RSX/overlays.h

```diff
@@ -11,17 +11,49 @@
 #include "Emu/Cell/Modules/cellSaveData.h"
 #include "Emu/Cell/Modules/cellMsgDialog.h"
 #include "Emu/Cell/Modules/sceNpTrophy.h"
+#include "Utilities/CPUStats.h"
+#include "Utilities/Timer.h"
 
 #include <time.h>
 
+template <>
+inline void fmt_class_string<perf_overlay_detail_level>::format(std::string& out, u64 arg)
```

Comment:
This function does not belong here imo. If perf_overlay_detail_level (which I feel should be a generic level-of-detail enum for any future options with incrementing levels named something like option_detail_level or something) is part of the core System, then it belongs in System.cpp. Keep in mind that RSX/Overlays is a completely optional part of rpcs3 afterall, like RSX/Capture and the three renderer backends with visual output. Non-core code is as much as possible kept out of the core emulator code, although sometimes it is difficult to do so cleanly.

Reply:
I know, originally I had the enum in the `perf_metrics_overlay` struct itself.

I _had_ to remove and rename it because of a cyclic dependency with `System` which I couldn't resolve any other way (`System.h` needed it for configuration, but `overlays.h` itself included `overlay_controls.h` which included `System.h`).

Maybe fixed with refactoring PR?

Reply:
Idk how it can be resolved easily either without using a static_cast and using a raw cfg::_int as the node base type. But I also mean this fmt method does not belong here if the enum is defined in the core. It would belong in System.cpp.
As for the enum, my idea was to promote it to core, but rename it from perf_xxx to a generic level of detail enum. That way it would make more sense if more options could be added which also use the same enum.